cdap-api/src/main/java/co/cask/cdap/api/schedule/Trigger.java

```diff
@@ -16,8 +16,79 @@
 
 package co.cask.cdap.api.schedule;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * A condition must be satisfied in order to trigger a schedule.
  */
 public interface Trigger {
+  /**
+   * Represents all known trigger types.
+   */
+  enum Type {
+    TIME("time"),
+    PARTITION("partition"),
+    STREAM_SIZE("stream-size"),
+    PROGRAM_STATUS("program-status"),
+    AND("and", true),
+    OR("or", true);
+
+    private static final Map<String, Type> CATEGORY_MAP;
```

Comment:
Why need this?? Just a plain enum is not enough?? It's only used programmatically to match the TriggerInfo type, right? Also, why not define it inside `TriggerInfo`?? Seems only used there.